creusot-contracts/src/std/slice.rs

```diff
@@ -100,6 +100,19 @@ extern_spec! {
         })]
         fn get<I : SliceIndexSpec<[T]>>(&self, ix: I) -> Option<&<I as SliceIndex<[T]>>::Output>;
 
+        #[requires(@mid <= (@self).len())]
+        #[ensures({
+            let (l,r) = result;  let sl = (@self).len();
+            ((@^self).len() == sl) &&
+            ((@l).len() == @mid && (@r).len() == sl - @mid) &&
+            ((@^l).len() == @mid && (@^r).len() == sl - @mid) &&
+            (forall<i:usize> @i < @mid
```

Comment:
could you use `Seq::subsequence` here? Like for [IteratorSpec](https://github.com/xldenis/creusot/blob/master/creusot-contracts/src/std/slice.rs#L172)

Reply:
Oh, indeed this is simpler. And then you don't even need to say much about the lengths.